core/src/main/java/cz/o2/proxima/scheme/AttributeValueAccessors.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2017-2021 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.scheme;
+
+import com.google.common.base.Preconditions;
+import cz.o2.proxima.annotations.Experimental;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Map;
+
+/** Classes providing access to Attribute values */
+@Experimental
+public class AttributeValueAccessors {
+
+  private AttributeValueAccessors() {}
+
+  /**
+   * Generic value accessor
+   *
+   * @param <T> value type
+   */
+  public interface GenericValueAccessor<T> extends Serializable {
+
+    /**
+     * Create value from object
+     *
+     * @param object create from
+     * @return value
+     */
+    default T createFrom(Object object) {
+      throw new UnsupportedOperationException("Method createFrom() is not implemented.");
+    }
+
+    /**
+     * Get value @TODO: this is a little bit weird
+     *
+     * @param value value
+     * @param <V> expected type
+     * @return value
+     */
+    @SuppressWarnings("unchecked")
+    default <V> V valueOf(T value) {
```

Comment:
should we move this into PrimitiveValueAccessor, so it does not leak to structure / array?